odoo/addons/base/models/ir_ui_view.py

```diff
@@ -482,7 +482,13 @@ def create(self, vals_list):
                     try:
                         if not values.get('arch') and not values.get('arch_base'):
                             raise ValidationError(_('Missing view architecture.'))
-                        values['type'] = etree.fromstring(values.get('arch') or values.get('arch_base')).tag
+                        text = values.get('arch_base')
+                        if isinstance(text, str) and re.search(r'<\?xml[^>]*encoding=.*?\?>', text, re.IGNORECASE):
+                            raise UserError(_(
+                                "Unicode strings with encoding declaration are not supported in XML.\n"
+                                "Remove the encoding declaration."
+                            ))
```

Comment:
```suggestion
                       validate_xml_encoding(text)
```
Import validate_xml_encoding from translate and use here